components/layout_thread/lib.rs

```diff
@@ -757,7 +757,7 @@ impl LayoutThread {
                 return false
             },
             Msg::SetNavigationStart(time) => {
-                self.paint_time_metrics.set_navigation_start(time);
+                (&self.paint_time_metrics).set_navigation_start(time);
```

Comment:
I made a trait ProgressiveWebMetric with function set_navigation_start, then ended up implementing it on &PaintTimeMetrics. I think I need to change the implementation to impl on PaintTimeMetrics and then make it so that you can call it on a ref